### PR TITLE
Basic cleanup

### PR DESCRIPTION
**Pull request by maririos**
- Removes Appconfiguration project
- Cleans using, unused variables and some string

### CODE REVIEW

File: sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <Description>This is the Microsoft Azure Key Vault Secrets client library</Description>
```
Comment (weshaggard) [inline sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj]: Why is this project not just including the `$(RequiredTargetFrameworks)` as the target frameworks?

Comment (weshaggard) [reply sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj]: lol.. nevermind I see it is but you have both TargetFramework and the TargetFrameworks property in this project. That also explains your CI failure. You should remove the TargetFramework property.

File: sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj [hunk 2]
```diff
@@ -5,7 +5,7 @@
...
     <AssemblyTitle>Microsoft Azure.Security.KeyVault.Secrets client library</AssemblyTitle>
     <Version>1.0.0-preview.1</Version>
-    <PackageTags>Microsoft Azure Application Configuration</PackageTags>
+    <PackageTags>Microsoft Azure Key Vault Secrets</PackageTags>
     <PackageReleaseNotes>
```
Comment (weshaggard) [inline sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj]: The TreatWarningsAsErrors, RootNamespace and AssembyName properties are redundant and can be removed.  I'd also check to see if the NoWarn and AllowUnsafeBlocks properties are necessary and if not remove them as well.

File: sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj (hunk 1 above)
Comment (maririos) [reply sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj]: Ohh yes! removing!

File: sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj (hunk 2 above)
Comment (maririos) [reply sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj]: After PR https://github.com/Azure/azure-sdk-for-net/pull/6123/files we want to have  `TreatWarningsAsErrors` enabled.
As for `NoWarn` and `AllowUnsafeBlocks ` removing it causes noises related to code in `Microsoft.Bcl.Json.Sources`

Comment (weshaggard) [reply sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj]: I thought I had enabled TreatWarningsAsErrors globally but I guess I didn't yet. I only enabled it in the build scripts https://github.com/Azure/azure-sdk-for-net/pull/5351. I'm fine with it staying for now but I will want to enable it globally at some point.

**Review by schaabs — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

File: sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj (hunk 2 above)
Comment (maririos) [reply sdk/keyvault/Azure.Security.Keyvault.Secrets/src/Azure.Security.Keyvault.Secrets.csproj]: sounds good. Happy to remove it later

**Pull request merged**